File: clients/android/demo/src/main/kotlin/com/clienttools/demo/model/LoginModels.kt
package com.clienttools.demo.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

// ── 请求体 ──────────────────────────────────────────────────────────────────

data class SendSmsRequest(val phone: String)
data class SmsLoginRequest(val phone: String, val code: String)
data class PasswordLoginRequest(val phone: String, val password: String)
data class EmailLoginRequest(val email: String, val password: String)

// ── 响应体 ──────────────────────────────────────────────────────────────────

data class BaseResponse(val code: Int, val message: String)

data class LoginResponse(
    val code: Int,
    val message: String,
    val data: LoginData?
)

data class LoginData(
    val token: String,
    val user: UserInfo
)

@Parcelize
data class UserInfo(
    val id: String,
    val name: String,
    val phone: String,
    val email: String,
    val avatar_url: String
) : Parcelable

File: clients/ios/demo/Sources/ClientToolsDemo/Login/LoginModels.swift
import Foundation

// MARK: - Requests

struct SendSmsRequest: Encodable {
    let phone: String
}

struct SmsLoginRequest: Encodable {
    let phone: String
    let code: String
}

struct PasswordLoginRequest: Encodable {
    let phone: String
    let password: String
}

struct EmailLoginRequest: Encodable {
    let email: String
    let password: String
}

// MARK: - Responses

struct BaseResponse: Decodable {
    let code: Int
    let message: String
}

struct LoginResponse: Decodable {
    let code: Int
    let message: String
    let data: LoginData?
}

struct LoginData: Decodable {
    let token: String
    let user: UserInfo
}

struct UserInfo: Codable {
    let id: String
    let name: String
    let phone: String
    let email: String
    let avatar_url: String
}
